test/cypress/integration/simplified-account/service-settings/worldpay-details/worldpay-details.cy.js

```diff
@@ -180,4 +189,151 @@ describe('Worldpay details settings', () => {
+          cy.get('input#username').should('have.value', '')
+          cy.get('input#password').should('have.value', '')
+        })
+
+        it('should redirect to the worldpay details landing page on valid form submission', () => {
+          setupStubs()
+
+          cy.visit(`/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/worldpay-details/one-off-customer-initiated`)
+
+          cy.get('input#merchantCode').type(VALID_MOTO_MERCHANT_CODE)
+          cy.get('input#username').type(VALID_WORLDPAY_USERNAME)
+          cy.get('input#password').type(VALID_WORLDPAY_PASSWORD)
+
+          cy.get('button#submitCredentials').click()
+
+          cy.location('pathname').should('eq', `/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/worldpay-details`)
+        })
+      })
+
+      describe('when credentials have been set', () => {
+        it('should populate the merchant code and username fields with the credentials', () => {
+          setupStubs({
+            credentials: {
+              one_off_customer_initiated: {
+                merchant_code: VALID_MOTO_MERCHANT_CODE,
+                username: VALID_WORLDPAY_USERNAME
+              }
+            }
+          })
+
+          cy.visit(`/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/worldpay-details/one-off-customer-initiated`)
+
+          cy.get('input#merchantCode').should('have.value', VALID_MOTO_MERCHANT_CODE)
+          cy.get('input#username').should('have.value', VALID_WORLDPAY_USERNAME)
+        })
+
+        // the password should not be returned by connector
+        // but this tests that even if it is, it will not be displayed
+        it('should not populate the password field', () => {
+          setupStubs({
+            credentials: {
+              one_off_customer_initiated: {
+                merchant_code: VALID_MOTO_MERCHANT_CODE,
+                username: VALID_WORLDPAY_USERNAME,
+                password: VALID_WORLDPAY_PASSWORD
+              }
+            }
+          })
+
+          cy.visit(`/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/worldpay-details/one-off-customer-initiated`)
+
+          cy.get('input#password').should('have.value', '')
+        })
+      })
+    })
+
+    describe('for a non-admin user', () => {
+      beforeEach(() => {
+        setupStubs({
+          role: 'view-and-refund'
+        })
+      })
+
+      it('should return a 403', () => {
+        cy.request({
+          url: `/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/worldpay-details/one-off-customer-initiated`,
+          failOnStatusCode: false
+        }).then(response => expect(response.status).to.eq(403))
+      })
+    })
+
+    describe('for a non-Worldpay account', () => {
+      beforeEach(() => {
+        setupStubs({
+          role: 'view-and-refund',
+          paymentProvider: STRIPE
+        })
+      })
+
```

Comment:
might want to test that the nav option doesn't show up for non worldpay accounts